src/protocols/secure_channel/CASESession.h

```diff
@@ -245,6 +245,9 @@ class DLL_EXPORT CASESession : public Messaging::UnsolicitedMessageHandler,
     static CHIP_ERROR HandleSigma3b(HandleSigma3Data & data, bool & cancel);
     CHIP_ERROR HandleSigma3c(HandleSigma3Data & data, CHIP_ERROR status);
 
+    // This will be called when Handle/Send Sigma3b returns error or we fail to schedule Handle/Send Sigma3c
+    void OnSigma3_AsyncFailure(CHIP_ERROR status);
```

Comment:
This file/class currently isn't using the "on" naming style for callbacks, in the names of functions, or in the names of the callbacks in WorkHelper. Also, it's not currently using underscores. I suggest following the naming conventions in this file and class. E.g. just `Sigma3AsyncFailure` or similar.

Reply:
I got motivated from HandleSigma2_and_SendSigma3 and I had a gut feeling to not use that convention, changing as per your suggestion.